omegaconf/base.py

```diff
@@ -377,28 +400,63 @@ def _resolve_interpolation(
         elif value_kind == ValueKind.STR_INTERPOLATION:
             value = _get_value(value)
             assert isinstance(value, str)
-            orig = value
-            new = ""
-            last_index = 0
-            for match in match_list:
-                new_val = self._resolve_simple_interpolation(
+            try:
+                # NB: `match_list` is ignored here as complex interpolations may be
+                # nested, the string will be re-parsed within `_evaluate_complex()`.
+                new = self._evaluate_complex(
+                    value,
                     key=key,
-                    inter_type=match.group(1),
-                    inter_key=match.group(2),
                     throw_on_missing=throw_on_missing,
                     throw_on_resolution_failure=throw_on_resolution_failure,
                 )
-                # if failed to resolve, return None for the whole thing.
-                if new_val is None:
-                    return None
-                new += orig[last_index : match.start(0)] + str(new_val)
-                last_index = match.end(0)
-
-            new += orig[last_index:]
+            except ParseError:
+                return None
             return StringNode(value=new, key=key)
         else:
             assert False
 
+    def _evaluate_simple(self, value: str, **kw: Any) -> Optional["Node"]:
```

Comment:
why **kw and not explicit parameters?

Reply:
I use this pattern sometime when there are arguments that are not actually used by a function, but need to be provided so that they can be forwarded to another function. It's similar to the typical `__init__(self, arg1, arg2, **kw)` used with `super().__init__(**kw)` in some places. Main benefit is to avoid modifying the code in a lot of places whenever you need to modify some of these arguments. There are drawbacks though and I know some people don't like it, so I'll just change it here (will also be more consistent with the rest of the code).

Reply:
Changed in 038f63c